### ISSUE TITLE
From cif to unit cell coordinates

### ISSUE THREAD

**Issue by Heisenburger2020** [biotite-dev/biotite#660]
Dear Sir/Madams,

I wonder if biotitie could calculate the unit cell configuration(not bio assembly) of protein from a mmcif file with the space group, unit cell shape and asymmetry unit in mmcif file?

Best,
Jiale Zhao

**Comment by padix-key** [comment]
No, unfortunately this information is currently only obtainable from PDB files via [`PDBFile.get_symmetry_mates()`](https://www.biotite-python.org/latest/apidoc/biotite.structure.io.pdb.PDBFile.html#biotite.structure.io.pdb.PDBFile.get_symmetry_mates).

The reason is that it is much simpler to compute them from PDB files as the necessary transformations are already available. In CIF files these transformations need to be computed from the space group (`_symmetry.space_group_name_H-M` field), which is much more complex and, to be honest, I lack the experience for it.

Therefore, if you (or someone who stumbles on this issue), have more experience in the space group to transformation matrix conversion, I would appreciate a PR here.

**Comment by Heisenburger2020** [comment]
Thank you for your reply. After several days of studying, I think I have understood how to get a unit cell transformation. I will try to PR.

**Comment by padix-key** [comment]
Great! I think the aim would be to have a function `get_symmetry_mates()` in https://github.com/biotite-dev/biotite/blob/main/src/biotite/structure/io/pdbx/convert.py, analogous to the function in `structure.io.pdb` (https://www.biotite-python.org/latest/apidoc/biotite.structure.io.pdb.get_symmetry_mates.html). But for the start some private function that takes the box vectors and the space group and returns a list of rotation matrices and translation vectors would be sufficient. After that I can support you, if you like to have some guidance.

**Comment by padix-key** [comment]
Hi, do you still work on this feature? Otherwise, I would try to tackle this issue soon.

**Comment by Heisenburger2020** [comment]
I am terribly sorry that I am not working on this feature anymore. I found another fantastic repo that can do this:
https://github.com/DanPorter/Dans_Diffraction/blob/4d3565e124435493fbc9498748519b57da0f04d5/Dans_Diffraction/classes_crystal.py#L1894
